omod-1.10/src/test/java/org/openmrs/module/attachments/rest/AttachmentRestController1_10Test.java

```diff
@@ -409,4 +463,82 @@ public void getAttachmentBytes_shouldDownloadFile() throws Exception {
 		Assert.assertEquals(downloadResponse.getHeader("File-Ext"), fileExtension);
 		
 	}
+	
+	@Test
+	public void doSearch() throws Exception {
+	
+		// Setup
+		String fileCaption = "Test file caption";
+		String uuid;
+		String mimeType = "application/octet-stream";
+		String fileExtension = "dat";
+		String fileName = "testFile." + fileExtension;
+
+		Patient patient = createBasicPatient();
+		MockMultipartHttpServletRequest uploadRequest = newUploadRequest(getURI());
+		MockMultipartFile file = new MockMultipartFile("file", fileName, mimeType, randomData);
+
+		uploadRequest.addFile(file);
+		uploadRequest.addParameter("patient", patient.getUuid());
+		uploadRequest.addParameter("fileCaption", fileCaption);
+
+		// Save attachment
+		deserialize(handle(uploadRequest));
+
+		// Search attachments for patient
+		MockHttpServletRequest request = request(RequestMethod.GET, getURI());
+		request.addParameter("patient", patient.getUuid());
+		SimpleObject response = deserialize(handle(request));
+		LinkedHashMap<String, String> result = (LinkedHashMap<String, String>) ((ArrayList<LinkedHashMap>) response
+		        .get("results")).get(0);
+
+		// Verify
+		Context.getPatientService().purgePatient(patient);
```

Comment:
Why would you purge the patient??

Reply:
I created this patient and wasn't sure the transaction will be rolled back after the test so I deleted it after I was done with it.

Reply:
It's no more. Since I'm now using a patient from the standard dataset, I don't have to worry about inconsistencies.